Assets/Scripts/ROS Scripts/ROS Connections v2/Drones/Matrice_ROSDroneConnection.cs

```diff
@@ -0,0 +1,846 @@
+                result = imu;
+                break;
+            case "/dji_sdk/rc":
+                remote_controller_msg = (parsed == null) ? new JoyMsg(raw_msg) : (JoyMsg)parsed;
+                result = remote_controller_msg;
+                break;
+            case "/dji_sdk/velocity":
+                Vector3Msg velocityMsg = (parsed == null) ? new Vector3Msg(raw_msg["vector"]) : (Vector3Msg)parsed;
+                velocity = new Vector3((float)velocityMsg.GetX(), (float)velocityMsg.GetY(), (float)velocityMsg.GetZ());
+                result = velocityMsg;
+                break;
+            case "/dji_sdk/height_above_takeoff":
+                relative_altitude = (parsed == null) ? (new Float32Msg(raw_msg)).GetData() : ((Float32Msg)parsed).GetData();
+                break;
+            case "/dji_sdk/local_position":
+                PointMsg pointMsg = (parsed == null) ? new PointMsg(raw_msg["point"]) : (PointMsg)parsed;
+                local_position = new Vector3(pointMsg.GetX(), pointMsg.GetY(), pointMsg.GetZ());
+                result = pointMsg;
+                Debug.Log(result);
+                break;
+            default:
+                Debug.LogError("Topic not implemented: " + topic);
+                break;
+        }
+        return result;
+    }
+
+    /// <summary>
+    /// Get ROS message type for a valid topic.
+    /// </summary>
+    /// <param name="topic"></param>
+    /// <returns></returns>
+    public string GetMessageType(string topic)
+    {
+        switch (topic)
+        {
+            case "/dji_sdk/attitude":
+                return "geometry_msgs/QuaternionStamped";
+            case "/dji_sdk/battery_state":
+                return "sensor_msgs/BatteryState";
+            case "/dji_sdk/flight_status":
+                return "std_msgs/UInt8";
+            case "/dji_sdk/gimbal_angle":
+                return "geometry_msgs/Vector3Stamped";
+            case "/dji_sdk/gps_health":
+                return "std_msgs/UInt8";
+            case "/dji_sdk/gps_position":
+                return "sensor_msgs/NavSatFix";
+            case "/dji_sdk/imu":
+                return "sensor_msgs/Imu";
+            case "/dji_sdk/rc":
+                return "sensor_msgs/Joy";
+            case "/dji_sdk/velocity":
+                return "geometry_msgs/Vector3Stamped";
+            case "/dji_sdk/height_above_takeoff":
+                return "std_msgs/Float32";
+            case "/dji_sdk/local_position":
+                return "geometry_msgs/PointStamped";
+        }
+        Debug.LogError("Topic " + topic + " not registered.");
+        return "";
+    }
+
+    /// <summary>
+    /// Disconnect the ros connection, terminated via ROSManager
+    /// </summary>
+    public void DisconnectROSConnection()
+    {
+        ros.Disconnect();
+    }
+
+
+    /// <summary>
+    /// Methods to execute service calls to the DJI SDK onboard the drone and corresponding methods to handle DJI SDK response
```

Comment:
I assume all the methods below are part of some interface implementation. 

If not (ie these are custom methods), please add individual summary comments.

Reply:
Nope, custom methods. Added documentation.